hammers/scripts/lease_stack_reaper.py

```diff
@@ -33,25 +33,28 @@ def __init__(self, user_id, name, email):
         self.email = email
         self.nodes = {}
 
-    def add_lease(self, node_id, lease_id, start_date, end_date, **kwargs):
+    def add_lease(self, node_type, lease_id, start_date, end_date, **kwargs):
         """Add lease to node list."""
-        if node_id not in self.nodes:
-            self.nodes[node_id] = []
+        if node_type not in self.nodes:
+            self.nodes[node_type] = []
 
-        self.nodes[node_id].append((lease_id, start_date, end_date))
+        # Data is by node id so do not add lease if it has already
+        # appeared for another node (i.e. multi-node leases).
+        if lease_id not in self.nodes[node_type]:
```

Comment:
I think if `self.nodes[node_type]` was a `set()` you wouldn't need this. I would expect Python can correctly identify duplicates on tuples of basic types without having to add any sort of equality operator.

Reply:
Also, I don't think this check will work given that you're only looking for `lease_id`.

```python
arr = [('a', 'b', 'c')]
print('a' in arr) # False
```

Reply:
You right. Set is actually called elsewhere on leases so nothing looked unusual.